barcode/writer.py

```diff
@@ -322,14 +322,15 @@ def _init(self, code):
         attributes = {"id": "barcode_group"}
         _set_attributes(group, **attributes)
         self._group = self._root.appendChild(group)
-        background = self._document.createElement("rect")
-        attributes = {
-            "width": "100%",
-            "height": "100%",
-            "style": f"fill:{self.background}",
-        }
-        _set_attributes(background, **attributes)
-        self._group.appendChild(background)
+        if self.background and self.background != (255, 255, 255, 0):
```

Comment:
I think it makes sense to exclude the background rectangle if `self.background is None`.

But I'm not convinced of excluding it if `self.background == (255, 255, 255, 0)`. Why exclude it instead of respecting the provided input?

Using `None` sounds like the correct approach for somebody who wants it entirely excluded.

Reply:
Actually this derives from my first usages: I wanted to have a transparent background, but using a 4-items tuple resulted in a black background. I did not go deeper in this (maybe there is a need to write `background = (255, 255, 255)` and then pass opacity 0), but definitely setting background to (255, 255, 255, 0) does not produce a transparent background.

I do, however, agree with you that this is more "sketchy", so maybe leaving only "None" as option and finding the correct way to manage transparency is better

Will you do the modification or I shall modify the PR?

Reply:
> setting background to (255, 255, 255, 0) does not produce a transparent background.

Do you have a quick example of this? Did you check the `background=` property on the SVG XML itself? Maybe it was the program that rendered it that didn't handle alpha properly?

> I do, however, agree with you that this is more "sketchy", so maybe leaving only "None" as option and finding the correct way to manage transparency is better
> 
> Will you do the modification or I shall modify the PR?

Please update the PR.

Reply:
> Do you have a quick example of this? Did you check the `background=` property on the SVG XML itself? Maybe it was the program that rendered it that didn't handle alpha properly?

Honestly I don't know where I got the hint to write `(255,255,255,0)`, since I did it a long time ago, but it appears that CSS (and consequently SVG) do not support that notation.

The library correctly wrote `fill:(255, 255, 255, 0)`, but this is wrong for two reasons: first you must write `rgb(r, g, b)` if you want to use three separate values, and then the fill parameter does not support alpha channel. The correct way to set transparency is using `fill-opacity`, so the style for the background rect shall be:

`fill:rgb(255, 255, 255);fill-opacity:0`

Up to now there is no way to set the fill-opacity parameter (maybe this can be added?), but with this PR maybe it will not be so much necessary.

In any case I'll update the PR